server.js

```diff
@@ -1,24 +1,105 @@
 import cors from "cors";
 import express from "express";
 import mongoose from "mongoose";
+import listEndpoints from "express-list-endpoints";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+const mongoUrl =
+  process.env.MONGO_URL || "mongodb://localhost/happy-thoughts-api";
 mongoose.connect(mongoUrl);
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
-const port = process.env.PORT || 8080;
+// Model
+const Thought = mongoose.model("Thought", {
+  text: {
+    type: String,
+    required: true,
+    minLength: 5,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+const port = process.env.PORT || 8081;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// GET /thoughts endpoint to return 20 results ordered by createdAt in descending order
+app.get("/thoughts", async (req, res) => {
+  try {
+    const thoughts = await Thought.find().sort({ createdAt: -1 }).limit(20);
+    res.json(thoughts);
+  } catch (error) {
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+});
+
+// GET /thoughts/oldest endpoint to return 20 oldest thoughts ordered by createdAt in ascending order
+app.get("/thoughts/oldest", async (req, res) => {
```

Comment:
Better to use query params for these types of queries. Then you'd only need one endpoint that's named after what it returns (/thoughts) and the user can choose whether or not they want to add a sorting query, i.e. /thoughts would work but /thoughts?sort=desc would also work